src/scanpy/plotting/_baseplot_class.py

```diff
@@ -221,17 +213,16 @@ def swap_axes(self, *, swap_axes: bool | None = True) -> Self:
         swap_axes
             Boolean to turn on (True) or off (False) 'swap_axes'. Default True
 
-
         Returns
         -------
         Returns `self` for method chaining.
 
         """
-        self.DEFAULT_CATEGORY_HEIGHT, self.DEFAULT_CATEGORY_WIDTH = (
-            self.DEFAULT_CATEGORY_WIDTH,
-            self.DEFAULT_CATEGORY_HEIGHT,
+        # TODO: this doesn’t make much sense
+        self.category_height, self.category_width = (
+            self.category_width,
+            self.category_height,
         )
```

Comment:
Agreed needs some investigating (what's the counter-factual when we remove it?)

Reply:
This just needs to be implemented in a way that actually respects the boolean parameter.